db/src/lib.rs

```diff
@@ -1,101 +1,9 @@
-use color_eyre::Result;
 pub use sqlx;
 use sqlx::postgres::PgPoolOptions;
 pub use sqlx::PgPool;
-use uuid::Uuid;
 
-#[derive(sqlx::FromRow)]
-pub struct User {
-    pub user_id: Uuid,
-    pub user_name: String,
-}
-
-#[derive(sqlx::FromRow, Debug)]
-pub struct Page {
-    pub page_id: Uuid,
-    pub user_id: Uuid,
-    pub url: String,
-}
-
-#[allow(clippy::module_name_repetitions)]
-#[derive(Debug)]
-pub enum AddUrlOutcome {
-    Created(Page),
-    Existing(Page),
-}
-
-pub async fn create_user(pool: &PgPool, user_name: &str) -> color_eyre::Result<Uuid> {
-    let result = sqlx::query!(
-        "INSERT INTO Users (user_name) VALUES ($1) RETURNING user_id",
-        user_name
-    )
-    .fetch_one(pool)
-    .await?;
-
-    Ok(result.user_id)
-}
-
-pub async fn add_url(
-    pool: &PgPool,
-    url: &str,
-    user_id: Uuid,
-    allow_existing: &bool,
-) -> color_eyre::Result<AddUrlOutcome> {
-    let new_page_id = Uuid::new_v4();
-
-    let upsert_result = sqlx::query_as!(
-        Page,
-        "INSERT INTO Pages (page_id, user_id, url) VALUES ($1, $2, $3)
-         ON CONFLICT (user_id, url) DO UPDATE SET url = EXCLUDED.url
-         RETURNING *",
-        new_page_id,
-        user_id,
-        url
-    )
-    .fetch_one(pool)
-    .await?;
-
-    if upsert_result.page_id == new_page_id {
-        println!("URL '{}' added successfully.", upsert_result.url);
-        Ok(AddUrlOutcome::Created(upsert_result))
-    } else if *allow_existing {
-        println!(
-            "URL '{}' exists but re-adding is allowed.",
-            upsert_result.url
-        );
-        Ok(AddUrlOutcome::Existing(upsert_result))
-    } else {
-        Err(color_eyre::eyre::eyre!(
-            "URL '{}' already exists and re-adding is not allowed.",
-            upsert_result.url
-        ))
-    }
-}
-
-pub async fn get_users(pool: &PgPool) -> Result<Vec<User>> {
-    let users = sqlx::query_as!(
-        User,
-        " 
-         SELECT user_id, user_name 
-         FROM Users
-         ",
-    )
-    .fetch_all(pool)
-    .await?;
-
-    Ok(users)
-}
-
-pub async fn get_username_by_id(
-    pool: &PgPool,
-    user_id: Uuid,
-) -> color_eyre::Result<Option<String>> {
-    let record = sqlx::query!("SELECT user_name FROM Users WHERE user_id = $1", user_id)
-        .fetch_optional(pool)
-        .await?;
-
-    Ok(record.map(|r| r.user_name))
-}
+pub mod add_url;
+pub mod users;
```

Comment:
calling the mod `add_url` & `users `in lib.rs

Reply:
Maybe the `add_urls` is just `urls` for the module name?